lib/active_model/serializer/fieldset.rb

```diff
@@ -18,14 +18,23 @@ def fields_for(type)
       attr_reader :raw_fields
 
       private
-
       def parsed_fields
+        # TODO: this is not very flexible.
+        # - we should probably support symbol, array, hash, etc
+        #
+        # JSONAPI::IncludeDirective supports these, but also changes the resulting structure.
+        # is it worth it to duplicate some of that functionality?
+        # or should IncludeDirective be modified to only allow one layer of options?
```

Comment:
> While working on that I've discovered an architectural problem that will require some careful unraveling

As I've been looking over everything the past week planning out my big refactor / unification, I've been realizing the whole of AMS is a giant bowl of spaghetti.